LeadKitAdditions/Sources/Services/ValidationService/ValidationService.swift

```diff
@@ -0,0 +1,104 @@
+import SwiftValidator
+import RxCocoa
+import RxSwift
+
+private enum ValidationServiceStateReactType {
+    case none
+    case all
+    case each
+}
+
+enum ValidationServiceState {
+    case initial
+    case valid
+    case invalid
+}
+
+extension ValidationServiceState {
+
+    var isValid: Bool {
+        return self == .valid
+    }
+
+}
+
+class ValidationService {
+
+    private var disposeBag = DisposeBag()
+
+    private(set) var validationItems: [ValidationItem] = []
+
+    private let stateHolder: Variable<ValidationServiceState> = Variable(.initial)
+    var state: ValidationServiceState {
+        return stateHolder.value
+    }
+    var stateObservable: Observable<ValidationServiceState> {
+        return stateHolder.asObservable()
+    }
+
+    private var validationStateReactType: ValidationServiceStateReactType = .none
+
+    func register(item: ValidationItem) {
+        register(items: [item])
+    }
+
+    func register(items: [ValidationItem]) {
+        validationItems += items
+        bindItems()
+    }
+
+    func unregisterAll() {
+        validationItems.removeAll()
+        bindItems()
+    }
+
+    func unregister(item: ValidationItem) {
+        unregister(items: [item])
+    }
+
+    func unregister(items: [ValidationItem]) {
+        items.forEach { item in
+            if let removeIndex = validationItems.index(where: { $0 === item }) {
+                validationItems.remove(at: removeIndex)
+            }
+        }
+
+        bindItems()
+    }
+
+    @discardableResult
```

Comment:
i don't think that we should remove this. This function validate all fields and publish result into stateHolder variable. And this return result is just one of ways to get info about validation